src/auth/strategies/google.strategy.ts

```diff
@@ -0,0 +1,36 @@
+import { PassportStrategy } from "@nestjs/passport";
+import { Strategy, VerifyCallback } from "passport-google-oauth20";
+import { config } from "dotenv";
+
+import { Injectable } from "@nestjs/common";
+
+config();
+
+@Injectable()
+export class GoogleStrategy extends PassportStrategy(Strategy, "google") {
+  constructor() {
+    super({
+      clientID: process.env.GOOGLE_CLIENT_ID,
+      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
+      callbackURL: "http://localhost:3000/api/v1/auth/google/redirect",
+      scope: ["email", "profile"],
+    });
+  }
+
+  async validate(
+    accessToken: string,
+    _refreshToken: string,
+    profile: any,
```

Comment:
You can type this as `Profile`

`import {  Profile } from "passport-google-oauth20";`

```suggestion
    profile: Profile,
```

Reply:
Yeah, will add this